### PR TITLE
Add Cloud Foundry Integration docs

### PR DESCRIPTION
**Pull request by kmshultz**
Based on the new Cloud Foundry tile, but with some more discussion and a few screenshots.

It's deploying to staging site now.

### CODE REVIEW

**Comment by kmshultz** [comment]
@irabinovitch let me know when you've got a nice screenboard screenshot and I'll add it in here.

(Or, we can merge this and add that later.)

**Comment by irabinovitch** [comment]
Lets merge without the screenshot for now.

File: content/integrations/cloud_foundry.md [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+
+Any Cloud Foundry deployment can send metrics and events to Datadog. The data helps you—the deployment administrator—track the health and availability of all nodes in the deployment, monitor the jobs they run, collect metrics from the Loggregator Firehose, and more.
+
+There are three points of integration with Datadog, each of which achieves a different goal:
+
+* **Datadog Agent BOSH release** — Install the Datadog Agent on every node in your deployment to track system, network, and disk metrics. Enable any other Agent checks you wish.
+* **Datadog Firehose Nozzle** — Deploy one or more Datadog Firehose Nozzle jobs. The jobs tap into your deployment's Loggregator Firehose and send all non-container metrics to Datadog.
+* **Datadog plugin for BOSH Health Monitor** — Configure your BOSH Director's Health Monitor to send heartbeats (as metrics) and alerts (as events) from each node's BOSH Agent to Datadog.
+
+If you do NOT administer a Cloud Foundry deployment—i.e. if you're an end user deploying applications onto a Cloud Foundry deployment—this page cannot help you.
```
Comment (irabinovitch) [inline content/integrations/cloud_foundry.md]: Lets rephrase this as "You must have administrative access to your CloudFoundry cluster in order to deploy these monitoring components."

File: content/integrations/cloud_foundry.md [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+
+As with the Datadog Agent, Datadog provides a BOSH release of the Datadog Firehose Nozzle. After uploading the release to your Director, you can add the Nozzle to an existing deployment, or create a new deployment that only includes the Nozzle. The instructions below assume you're adding it to an existing Cloud Foundry deployment that has a working Loggregator Firehose.
+
+### Upload Datadog's release to your BOSH Director
+
+~~~
+# BOSH CLI v1
+bosh upload release https://github.com/DataDog/datadog-firehose-nozzle-release/releases/download/61/datadog-firehose-nozzle-release-61.tgz
+
+# BOSH CLI v2
```
Comment (irabinovitch) [inline content/integrations/cloud_foundry.md]: Do we need to update these URLs each time we release? Do we need to do the same in Dogweb? @gmmeyer @kmshultz Is there a permanent URL we can offer so we dont have to remember to upload multiple page.

File: content/integrations/cloud_foundry.md [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+      subscription_id: datadog-nozzle  # can be anything (firehose streams data evenly to all jobs using the same subscription_id)
+      # disable_access_control: true   # for development only
+      # insecure_ssl_skip_verify: true # for development only; enable if your UAA does not use a verifiable cert
+    uaa:
+      client: datadog-firehose-nozzle # client name you just configured
+      client_secret: <SECRET_YOU_JUST_CONFIGURED>
+      url: <UAA_URL> # e.g. https://uaa.your-cf-domain.com:8443
+~~~
+
+You can set `metrics_prefix` to anything you like, but if you want to use the default Cloud Foundry screenboard and to see metrics metadata in dashboards and graphs, set it to `cloudfoundry.nozzle.`.
```
Comment (irabinovitch) [inline content/integrations/cloud_foundry.md]: Lets discourage this. It has billing implications if we let them choose anything by cloudfoundry.nozzle as the prefix

File: content/integrations/cloud_foundry.md [hunk 4]
```diff
@@ -0,0 +1,283 @@
...
+      datadog_enabled: true
+      datadog:
+        api_key: <YOUR_DATADOG_API_KEY>
+        application_key: <YOUR_DATADOG_APP_KEY>
+        pagerduty_service_name: "your_service_name" # optional
+~~~
+
+### Redeploy the Director
+
+BOSH cannot simply configure the plugin and restart the Health Monitor; it must redeploy the Director entirely. **To do this without without fully wiping the Director—including its persistent disks and its PostgreSQL database—you MUST have a `state.json` (or similarly named) file that accurately reflects the current state of your Director.**
```
Comment (irabinovitch) [inline content/integrations/cloud_foundry.md]: Can we confirm this with the CF team? This text seems like it might scare off users.

File: content/integrations/cloud_foundry.md (hunk 3 above)
Comment (kmshultz) [inline content/integrations/cloud_foundry.md]: If we want to discourage it, we should not show the option in this snippet at all. Problem is, the default value is `cf.nozzle`, but we can change that in the nozzle release. What do you think @gmmeyer?

File: content/integrations/cloud_foundry.md (hunk 4 above)
Comment (kmshultz) [inline content/integrations/cloud_foundry.md]: This is my understanding from the [BOSH docs](https://bosh.io/docs/cli-envs.html#deployment-state), but I haven't confirmed that for sure it will destroy ones disks.

If what is written there in bold is true, then it *should* scare users. Would we rather have weaker language that results in someone unwittingly blowing away their Director's database?

File: content/integrations/cloud_foundry.md (hunk 1 above)
Comment (kmshultz) [inline content/integrations/cloud_foundry.md]: How about:

"These integrations help administrators of Cloud Foundry deployments, not end users."

I like to finish Overviews by saying whom the integration is _not_ for, if there's any chance for confusion. In this case, maybe it will be obvious to CF cluster end users that it's not for them.

And, if we say 'end users, this isn't for you', then we should point them to what is for them (I guess DogStatsD docs).

File: content/integrations/cloud_foundry.md (hunk 2 above)
Comment (kmshultz) [inline content/integrations/cloud_foundry.md]: That'd be no good for sure, we should provide an unchanging URL.

Comment (gmmeyer) [inline content/integrations/cloud_foundry.md]: There will be one once we get it uploaded to s3. I'm working on that. it'll be something like:

https://cloudfoundry.datadoghq.com/datadog-firehose-nozzle/datadog-firehose-nozzle-release-latest.tgz

File: content/integrations/cloud_foundry.md (hunk 3 above)
Comment (gmmeyer) [inline content/integrations/cloud_foundry.md]: I believe I've changed the default value to cloudfoundry.nozzle

Comment (kmshultz) [inline content/integrations/cloud_foundry.md]: Ok, cool, looks like you did: https://github.com/DataDog/datadog-firehose-nozzle-release/pull/5

Should we omit the option here?

File: content/integrations/cloud_foundry.md [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+title: Datadog-Cloud Foundry Integration
+integration_title: Cloud Foundry
+kind: integration
+git_integration_title: cloud_foundry
+newhlevel: true
+beta: true
+---
+# Overview
+
+Any Cloud Foundry deployment can send metrics and events to Datadog that help you track the health and availability of all nodes in the deployment, monitor the jobs they run, collect metrics from the Loggregator Firehose, and more.
```
Comment (irabinovitch) [inline content/integrations/cloud_foundry.md]: This feels like a bit of a run on.  Can we break this into 2 sentences?

File: content/integrations/cloud_foundry.md [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+      use_dogstatsd: yes
+      dogstatsd_port: 18125               # Many CF deployments have a StatsD already on port 8125
+      api_key: <YOUR_DATADOG_API_KEY>
+      tags: ["cloudfoundry_deployment_1"] # any tags you wish
+      generate_processes: true            # to enable the process check
+~~~
+
+To see which `datadog-agent` release version you uploaded earlier, run `bosh releases`.
+
+If you don't have a local copy of the runtime configuration, get it from the Director (`bosh runtime-config`) and add the above to it. If the Director's runtime configuration is empty, add the above to a new `runtime.yml`.
```
Comment (irabinovitch) [inline content/integrations/cloud_foundry.md]: If you don't have a local copy of the runtime configuration, it can be retrieved from the Director (`bosh runtime-config`). Once retrieved add the above addon configuration to it with your account credentials. If the Director's runtime configuration is empty, create a new `runtime.yml` and add the Datadog configuration there.

File: content/integrations/cloud_foundry.md (hunk 5 above)
Comment (kmshultz) [inline content/integrations/cloud_foundry.md]: Ok. Yeah, it was originally two sentences and I broke it into two for some reason. I'll change it to the original.
